src/main/java/com/openelements/conduct/integration/slack/SlackIntegration.java

```
@@ -0,0 +1,99 @@
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import org.jspecify.annotations.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SlackIntegration implements ResultHandler {
+
+    private final static Logger log = LoggerFactory.getLogger(SlackIntegration.class);
+
+    private final MethodsClient slackClient;
+    private final String channelId;
+
+    public SlackIntegration(@NonNull String slackToken, @NonNull String channelId) {
+        Objects.requireNonNull(slackToken, "slackToken must not be null");
+        this.channelId = Objects.requireNonNull(channelId, "channelId must not be null");
+        if (channelId.isBlank()) {
+            throw new IllegalArgumentException("channelId must not be empty");
+        }
+        log.info("Initializing Slack integration");
+        try {
+            Slack slack = Slack.getInstance();
+            this.slackClient = slack.methods(slackToken);
+            
+            // Send initialization notification
+            sendInitializationMessage();
+            log.info("Slack integration initialized successfully");
+        } catch (Exception e) {
+            log.error("Failed to initialize Slack integration", e);
+            throw new RuntimeException("Failed to initialize Slack integration", e);
+        }
+    }
+    
+    private void sendInitializationMessage() {
+    try {
+        ChatPostMessageRequest request = ChatPostMessageRequest.builder()
+                .channel(channelId)
+                .text("🚀 *Conduct Guardian* has been successfully connected to this Slack channel. " +
+                      "Code of conduct violations will be reported here.")
+                .build();
+
+        var response = slackClient.chatPostMessage(request);
+        if (!response.isOk()) {
+            log.error("Slack API Error: {}", response.getError());  // <-- this is key
```

We should throw an exception for this case